packages/mds-stream/stream-kafka.ts

```diff
@@ -0,0 +1,116 @@
+import { VehicleEvent, Telemetry, Device } from '@mds-core/mds-types'
+import log from '@mds-core/mds-logger'
+import Kafka, { ProducerStream } from 'node-rdkafka'
+import { Stream } from './stream-interface'
+
+const {
+  env: { KAFKA_HOST = 'localhost:9092' }
+} = process
+
+/*
+  FIXME:
+  Necessary interface due to already-fixed typedef problem in the node-rdkafka repo,
+  will be removed upon next upstream release
+*/
+interface Producer {
+  createWriteStream(conf: any, topicConf: any, streamOptions: any): ProducerStream
+}
+
+/*
+  FIXME:
+  Strange casting here necessary due to problem described in the Producer interface above
+*/
+const eventStream = ((Kafka.Producer as unknown) as Producer).createWriteStream(
+  {
+    'metadata.broker.list': KAFKA_HOST,
+    'queue.buffering.max.messages': 100000
+  },
+  {},
+  {
+    topic: 'mds.event'
+  }
+)
+
+/*
+  FIXME:
+  Strange casting here necessary due to problem described in the Producer interface above
+*/
+const telemetryStream = ((Kafka.Producer as unknown) as Producer).createWriteStream(
+  {
+    'metadata.broker.list': KAFKA_HOST,
+    'queue.buffering.max.messages': 100000
+  },
+  {},
+  {
+    topic: 'mds.telemetry'
+  }
+)
+
+/*
+  FIXME:
+  Strange casting here necessary due to problem described in the Producer interface above
+*/
+const deviceStream = ((Kafka.Producer as unknown) as Producer).createWriteStream(
+  {
+    'metadata.broker.list': KAFKA_HOST,
+    'queue.buffering.max.messages': 100000
+  },
+  {},
+  {
+    topic: 'mds.device'
+  }
+)
+
+eventStream.on('error', async err => {
+  await log.error(`Kafka Error ${JSON.stringify(err)}`)
+})
+
+telemetryStream.on('error', async err => {
+  await log.error(`Kafka Error ${JSON.stringify(err)}`)
+})
+
+deviceStream.on('error', async err => {
+  await log.error(`Kafka Error ${JSON.stringify(err)}`)
+})
+
+const writeTelemetry = (telemetry: Telemetry[]): Promise<void> => {
+  const results = telemetry.map(telem => telemetryStream.write(JSON.stringify(telem)))
+  const failures = results.filter(result => !result).length
+  // eslint-disable-next-line promise/avoid-new
+  return new Promise((resolve, reject) => {
+    if (failures !== 0) {
+      reject(failures)
+    }
+    resolve()
+  })
```

Comment:
Are the stream writes synchronous? Could make the method `async` and then just throw when failures !== 0 instead of new'ing up a promise. Also, could just use `Promise.reject(failures)` or `Promise.resolve()`

Reply:
The stream writes are indeed synchronous, I'm gonna follow Promise.xxx patterns for now. Trivial to refactor once we can clean up that NATS/Redis streams client code a bit.